sky/resources.py

```diff
@@ -307,6 +312,23 @@ def __init__(
             disk_tier = resources_utils.DiskTier(disk_tier_str)
         self._disk_tier = disk_tier
 
+        if isinstance(network_tier, str):
+            network_tier_str = str(network_tier).lower()
+            supported_tiers = [
+                tier.value for tier in resources_utils.NetworkTier
+            ]
+            if network_tier_str not in supported_tiers:
+                with ux_utils.print_exception_no_traceback():
+                    raise ValueError(
+                        f'Invalid network_tier {network_tier_str!r}. '
+                        f'Network tier must be one of '
+                        f'{", ".join(supported_tiers)}.')
+            network_tier = resources_utils.NetworkTier(network_tier_str)
+        self._network_tier = network_tier
+        if self._network_tier == resources_utils.NetworkTier.BEST:
+            if isinstance(self._cloud, clouds.GCP):
+                self._image_id = {self._region: _NETWORK_GCP_IMAGE_ID}
```

Comment:
We should have a comment here for why we switch to the image. Also, on the constant above.

Reply:
Actually, what if the user also specifies a image? I am wondering for this kind of cloud specific handling. Would it be possible to have it inside the `clouds.GCP`, instead of leaking the abstraction in the `Resources` class.

Reply:
I updated the image in `clouds.GCP`